src/worker/mod.rs

```diff
@@ -275,35 +278,44 @@ impl Worker {
     pub(crate) async fn complete_activity(
         &self,
         task_token: TaskToken,
-        status: activity_result::Status,
+        status: activity_execution_result::Status,
     ) -> Result<(), CompleteActivityError> {
         if task_token.is_local_activity_task() {
-            if let Some(la_info) = self.local_act_mgr.complete(&task_token) {
-                if let Err(e) = self
-                    .wft_manager
-                    .notify_of_local_result(
-                        &la_info.workflow_execution.run_id,
-                        la_info.seq,
-                        LocalResolution::LocalActivity {
-                            result: ActivityResult {
-                                status: Some(status),
+            let as_la_res: LocalActivityExecutionResult = status.try_into()?;
+            match self.local_act_mgr.complete(&task_token, &as_la_res) {
+                LACompleteAction::Report(LocalInFlightActInfo {
+                    la_info,
+                    dispatch_time,
+                    ..
+                }) => {
+                    if let Err(e) = self
+                        .wft_manager
+                        .notify_of_local_result(
+                            &la_info.workflow_exec_info.run_id,
+                            la_info.schedule_cmd.seq,
+                            LocalResolution::LocalActivity {
+                                result: as_la_res,
+                                runtime: dispatch_time.elapsed(),
                             },
-                            runtime: la_info.dispatch_time.elapsed(),
-                        },
-                    )
-                    .await
-                {
-                    error!(
-                        "Problem completing local activity {}: {:?} -- will evict the workflow",
-                        task_token, e
-                    );
-                    self.request_wf_eviction(
-                        &la_info.workflow_execution.run_id,
-                        "Issue while completing local activity",
-                    );
+                        )
+                        .await
+                    {
+                        error!(
+                            "Problem completing local activity {}: {:?} -- will evict the workflow",
+                            task_token, e
+                        );
+                        self.request_wf_eviction(
+                            &la_info.workflow_exec_info.run_id,
+                            "Issue while completing local activity",
+                        );
+                    }
+                }
+                LACompleteAction::WillBeRetried => {
+                    // Nothing to do here
+                }
+                LACompleteAction::Untracked => {
+                    error!("Tried to complete untracked local activity {}", task_token);
```

Comment:
Should this error just be logged?
Is this "error" severity? It would probably happen if user ignores cancellation in their activity implementation.

Reply:
You mean should it do more than just log? All it does is log now.

We can change it to warn when working on cancels.

Reply:
I'd maybe just warn because the case I stated above

Reply:
I'll do it when working on cancels